src/main/java/it/aboutbits/springboot/emailservice/lib/model/Email.java

```diff
@@ -0,0 +1,82 @@
+package it.aboutbits.springboot.emailservice.lib.model;
+
+import it.aboutbits.springboot.emailservice.lib.EmailState;
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.NamedAttributeNode;
+import jakarta.persistence.NamedEntityGraph;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.Table;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.JdbcTypeCode;
+import org.hibernate.annotations.UpdateTimestamp;
+import org.hibernate.type.SqlTypes;
+
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Set;
+
+@NamedEntityGraph(
+        name = "email_service_emails-entity-graph",
+        attributeNodes = {
+                @NamedAttributeNode("attachments")
+        }
+)
+@Entity
+@Getter
+@Setter
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+@Table(name = "email_service_emails")
+public class Email {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Enumerated(EnumType.STRING)
+    private EmailState state;
+
+    private String subject;
+
+    private String fromAddress;
+    private String fromName;
+
+    @JdbcTypeCode(SqlTypes.JSON)
+    private List<String> recipients;
+
+    private String textBody;
+    private String htmlBody;
+
+    @OneToMany(cascade = CascadeType.PERSIST, mappedBy = "email", orphanRemoval = true, fetch = FetchType.EAGER)
+    private Set<EmailAttachment> attachments;
+
+    private String reference;
```

Comment:
Since the email service does not know about your application, this is an optional reference the application can use to create a relation. Like: what caused the sending of that email?

Reply:
I would personally remove it and let the application that uses the email service keep a reference to the email by the emails id.